src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java

```diff
@@ -208,6 +211,17 @@ public void testGenomicsDBThreeLargeSamplesSitesOnlyQuery() throws IOException {
         testGenomicsDBImporterWithGenotypes(LOCAL_GVCFS, intervals, COMBINED_SITES_ONLY, b38_reference_20_21, true, true, true);
     }
 
+    @Test(expectedExceptions={UserException.BadInput.class, CompletionException.class}, expectedExceptionsMessageRegExp=".*GenomicsDBImport does not support GVCFs.*")
```

Comment:
Why do we expect a `CompletionException` here?

Reply:
I didn't really understand if it was a gradle thing or a GATK/GenomicsDB thing but the code is being run in some async setup, and what I'm actually getting back is a CompletionException which contains the BadInput exception.  If you'd rather I can catch the CompletionException in the test and rethrow the cause?

Reply:
@tfenne It looks like you've uncovered a problem in the tool itself. In `GenomicsDBImport.traverse()`, we need to add an additional catch block for `CompletionException` around the `importer.executeImport()` call (The `CompletionException` is being caused by a `future.join()` call deep in the GenomicsDB library). Once you do this, you can remove `CompletionException` from the list of `expectedExceptions` in your test. 

In order to ensure that your MNP exception message will actually be shown to the user, I'd also suggest unpacking the `CompletionException` in that catch block in `GenomicsDBImport` if `e.getCause() instanceof UserException`, and rethrowing `e.getCause()` in that case. It's gross, but I think making the error message visible to the user should take priority here. 

This also made me realize that you should have a second integration test that exercises the `getFeatureReadersInParallel()` path in the tool. You can do this by copying your existing test, adding an overload of `testGenomicsDBImporter()` that takes a `threads` argument and passes it through to `writeToGenomicsDB()` as the final arg, and calling it with threads = 2.

Reply:
Done.  I deviated a little from what you suggested:

1. In `GenomicsDBImport` if the cause of the `CompletionException` is a runtime exception I rethrow that, otherwise the original exception.  I figure it's always going to be more informative to see the actual exception where possible.
2. I didn't overload `testGenomicsDBImporter()`, I just added a `threads` parameter and in the other ~3 places passed `1`.
3. In my actual test, I just added a `for` loop that runs the same code twice with threads = [1, 2].

My thoughts being that (2) and (3) keep things a little more compact rather than duplicating or chaining code.